spec/lib/s3_inventory_spec.rb

```diff
@@ -162,6 +126,58 @@
     expect(Upload.by_users.order(:url).pluck(:url, :etag)).to eq(files)
   end
 
+  context "when site was restored from a backup" do
+    before do
+      freeze_time
+      BackupMetadata.update_last_restore_date(Time.now)
+    end
+
+    it "should run if inventory files are at least #{described_class::WAIT_AFTER_RESTORE_DAYS.days} days older than the last restore date" do
+      client.stub_responses(
+        :list_objects_v2,
+        {
+          contents: [
+            {
+              key: "symlink.txt",
+              last_modified:
+                BackupMetadata.last_restore_date + described_class::WAIT_AFTER_RESTORE_DAYS.days,
+              size: 1,
+            },
+          ],
+        },
+      )
+
+      client.expects(:get_object).once
```

Comment:
Don't want to block you, but I was tracing how this side effect happens.

When we call `backfill_etags_and_list_missing`, from what I'm seeing here we invoke `get_object` somewhere but I can't actually find it.

Reply:
O it happens in the following order 

https://github.com/discourse/discourse/blob/a4c5f85b10ea04f57ef5b248d69b44b5797471a3/lib/s3_inventory.rb#L271

https://github.com/discourse/discourse/blob/a4c5f85b10ea04f57ef5b248d69b44b5797471a3/lib/s3_helper.rb#L277

https://github.com/discourse/discourse/blob/a4c5f85b10ea04f57ef5b248d69b44b5797471a3/lib/s3_helper.rb#L259

I know it isn't clear but the alternative here is to not stub responses from S3 and instead stub all the outgoing requests. The trouble with that approach is that we have to form the request's response by hand.

Reply:
Ahhh I see my folly.. I was looking for `get_object`.